### PR TITLE
Weapon and Tool Additions

### PR DESCRIPTION
**Pull request by chaosvolt**
Rivet suggested I mainline these ideas I had. Adds 2 new tools, changing
the relevance of 1 tool, and a few added guns.
1. Pliers, basic bolt-turning item and therefore inferior to the wrench.
2. A SET of screwdrivers, providing the fine screwdriving ability. More
   or less common than the basic screwdriver depending on where you look.
   This means the basic screwdriver no longer gives fine screwdriving
   quality.
3. Sten 9mm SMG. No longer craftable.
4. Thompson SMG, in .45 ACP.
5. M1903 Springfield, in .30-06.
6. M1 Garand, in .30-06. Sorry, no ping noise for you.
7. Browning Automatic Rifle, in .30-06.
8. M60 GPMG, in 7.62x51mm.
9. M240 GPMG/MMG, in 7.62x51mm.

These guns need a good hard look to determine prices and stats relative
to other weapons.

### CODE REVIEW

**Comment by chaosvolt** [comment]
Now, I've been up past 3:30, sleeeep. @_@

**Comment by kevingranade** [comment]
Stein needs more than, "someone historically could craft it" for
inclusion.  Using what tools?  What materials?  If we're talking a lathe
and a metal press, the survivor doesn't have those.

**Comment by Rivet-the-Zombie** [comment]
Yeah, I've got a .pdf of the original blueprints for a Sten mk. II on my computer, and I can say for certain that without a halfway-decently equipped metal shop, you're not going to be able to make one of those; on the other hand, if you do have access to a halfway-decently equipped metal shop, it looks like it would take virtually no specialized skill to make.

**Comment by DavidKeaton** [comment]
Rivet, the Sten ain't the grease gun, right? Cuz that was mostly rolled metal.

**Comment by Sirbab** [comment]
Looking at the mechanism for the stem 9mm, the mechanism is rather simple, but that's not to say someone would be able to make it without decent fabrication tools. See: http://upload.wikimedia.org/wikipedia/commons/8/82/StenmkII_animation.gif in terms of cataclysm difficulty, i'd say a "bit" more difficult than the flintlock rifle, certainly would take a hell of a lot more time to fabricate. pretty cool gun though, the blowback is merely arrested from the spring sofar as i can tell.

**Comment by Rivet-the-Zombie** [comment]
>Rivet, the Sten ain't the grease gun, right?

Correct. Though they have a similar appearance, the Sten and the M3 are very different firearms.

**Comment by DavidKeaton** [comment]
the m3 should be relatively craft-able then, yeah?

**Comment by Rivet-the-Zombie** [comment]
The M3's guts are a bit more complicated than the Sten's, but the primary thing that makes them unfeasible to craft as it currently stands is that you'd need a lathe, press, shear, and metal forming wheel. If the player had access to a metal shop, then producing simple SMGs like those would really only require blueprints, material, and the skill and knowledge of how to operate the machinery to bring it all together.

**Comment by chaosvolt** [comment]
My apologies for the substantial delays, had a long day.

So, should I remove the Sten recipe, or bump up the requirements? Any other items in need of tweaks?

**Comment by chaosvolt** [comment]
Now, if I had any good data on dispersion, recoil, and price for these guns. Only instance where I could find a reliable figure for pricing was for the Thompson, with auction result ranges having been mentioned on the weapon's wikipedia page.

**Comment by DavidKeaton** [comment]
Ah, fair enough. I could see the addition of a metalsmith building being able to produce such things, but those tools won't be in an RV feasibly. :-) so if I read Rivet right (thanks for the info btw!), it may be best to remove the ability to make the Stew altogether.

**Comment by chaosvolt** [comment]
I would not want to make 9mm Stew. o3o

But yeah, unless the vehicle welding rig can make up for it, I doubt we have the tools for it.

**Comment by DavidKeaton** [comment]
I changed it to Sten after the fact. :-) hehe yeah the tools needed won't be on the rig. ;-)

**Comment by chaosvolt** [comment]
I blame typo imps. <3

Also, the fanciest gun you can make is the Survivor's .223 Carbine, if I recall.

**Comment by DavidKeaton** [comment]
true, but that's a makeshift weapon. :-)

**Comment by chaosvolt** [comment]
Trying to recall if it's meant to be an autoloader or a manual action. If it's semi-auto that sets a precedent. o3o

**Comment by DavidKeaton** [comment]
doesn't mean it's a revolver, ever seen a lever action?

**Comment by chaosvolt** [comment]
Lever actions don't eject all their casings out at once, and revolvers don't generally have tube magazines. Nevermind the fact that tube magazine, plus centerfire primer, plus spitzer bullet equals bad idea. :V

**Comment by DavidKeaton** [comment]
not saying it ejects all casings, but I did misread the "eject after reload" to be "eject after firing", so the onus of the misunderstanding is mine lol.

**Comment by DavidKeaton** [comment]
however, the statement "revolvers *don't generally* have tube magazines" is evidence of a makeshift gun. a survivor weapon, if you will. ;-)

**Comment by chaosvolt** [comment]
True. Definitely a rather weird little thing though.

**Comment by chaosvolt** [comment]
Also, I still would love .357 Magnum. Any idea how one could implement that ammuniton type overlap idea? Anyone posted it as a PR or issue yet? owo

File: data/json/item_groups.json [hunk 1]
```diff
@@ -2747,6 +2753,8 @@
       ["mosin91_30_ebr", 1],
       ["mosin44_ebr", 1],
       ["sks", 6],
+      ["m1903", 5],
```
Comment (kevingranade) [inline data/json/item_groups.json]: Neither of these is going to be common in a modern rifle collection, dial the weight down to 1 or so, probably the same for all the instances of them.

**Comment by DavidKeaton** [comment]
was about to say the Judge is a perfect example (great gun to fire btw). albeit a better idea is to not define "Judge Usable", but to define a gun as having multiple ammo types. in all honesty it wouldn't take all too much code.

File: data/json/items/tools.json [hunk 2]
```diff
@@ -5790,6 +5790,30 @@
...
+    "qualities": [["SCREW", 1]]
+  },
+
+  {
+    "id": "screwdriver_set",
+    "type": "TOOL",
+    "symbol": ";",
+    "color": "yellow",
+    "name": "screwdriver set",
+    "description": "This is a set of screwdrivers in several sizes and blade types.  Guaranteed to have the right tools for more precise work.",
```
Comment (kevingranade) [inline data/json/items/tools.json]: Woo, this is exactly where I was planning on going with high-quality screwdriver tools.
Now need same for spanners, and a mechanism for collecting enough misc screwdrivers to make a set :3

Comment (Rivet-the-Zombie) [inline data/json/items/tools.json]: Yes, this is very good!

**Comment by DavidKeaton** [comment]
sadly, I wanted that pinging noise. :-(

**Comment by chaosvolt** [comment]
Meep. Guess I could tone down the rarity for the WW2-era guns, sure. And thanks, was hoping the screwdriver set would seem about right.

And @DavidKeaton, I would love the ping too, but no idea how to implement that, code-wise. owo

**Comment by DavidKeaton** [comment]
well for the ammo bit, it would be better to do it on the weapon side so you wouldn't have to change it later. as far as the Garand noise, iirc it has to do with the rifle itself making the noise. in the code, you could simply add an if statement to the reload function as a quick hack.

**Comment by chaosvolt** [comment]
Ah, right. Wasn't sure which way to code ammo overlap would be simpler, so if someone can figure out how to do either method... owo

And yeah, might save the Garand pinging/clanging for after the M1 becomes a thing. o3o

**Comment by chaosvolt** [comment]
Also, the capacities for the M240 and M60 were mostly guesswork as I wasn't sure what sorts of belt boxes are used when they aren't being hand-fed by a loader linking the belts. Checking the FMs I have online copies of to see if they have info on this, so far it seems 100 round bandoliers might be the norm for the M60. O.o

**Comment by Rivet-the-Zombie** [comment]
>as far as the Garand noise, iirc it has to do with the rifle itself making the noise. in the code, you could simply add an if statement to the reload function as a quick hack.

I don't think this is enough of a thing to warrant bothering; when you're actually firing one you don't really even notice it unless you're specifically listening for it - the report is loud enough that the sound of the action is lost in a giant **boom**.

**Comment by chaosvolt** [comment]
True. After all, my first hint of ever suggesting a Garand ping was in the Bad Ideas thread, so that speaks for itself. o3o

**Comment by DavidKeaton** [comment]
@Rivet-the-Zombie, point. 30-06 "ought" do that with its sound. ;-)

**Comment by chaosvolt** [comment]
I'm gonna eat you for that pun. =w=

**Comment by DavidKeaton** [comment]
mwuahaha

**Comment by chaosvolt** [comment]
Nyom. o3o

Anyway, any further ideas for additions?

**Comment by Chezzo** [comment]
Wasn't the Sten gun a horrible weapon that jammed all the time?

Anyways, you can craft it, according to Practical Scrap Metal Arms Vol. 3:

http://www.mediafire.com/view/5v4852yage8d8c5

**Comment by chaosvolt** [comment]
I might see how much I should reduce the durability to account for that. And does that link permit crafting with the sort of hand tools a survivor would have? Remember metal shops aren't a thing yet. owo

**Comment by DavidKeaton** [comment]
can't think of any additions!

**Comment by chaosvolt** [comment]
Yay, that's a good sign. owo

**Comment by chaosvolt** [comment]
> Could you slip in saber, not fencing more cavalry or pirate cutlass, for little ole' me? Or better yet both? Please been begging for a while and somehow no dice. 

Hmm. Any suggestions of stats on such a thing? =w=

**Comment by DavidKeaton** [comment]
I'd be okay with this. Considering it is curved towards the end, the cutting would be decent. More than a sword, but less than what a scimitar would be.

**Comment by chaosvolt** [comment]
Hopefully it looks alright stat-wise, then.

**Comment by Chezzo** [comment]
Yeah, ChaosVolt. That magazine I linked to makes a sten gun with an angle grinder, a hacksaw, a drill, a stand welder, a dremel and some hand files. 

It is a really good magazine, too. I like to know how to make guns out of tin cans and bits of string.

**Comment by chaosvolt** [comment]
Hmm. I did kinda think that the only real niche the Sten fills, due to already having a decent selection of 9mm SMGs, was the possibility of crafting it.

A drill, hacksaw, and welder we obviously can obtain in-game already. Not sure what tools exist that would allow replacing the angle grinder, dremel tool, and files though. owo

Anyone have suggestions on implementing this? Do note we can already make a semi-automatic .223 rifle in the game, when in real-life making a full-auto-only gun is simpler in terms of engineering.

**Comment by kevingranade** [comment]
Crafting automatic weapons in mainline is out, full stop.  Feel free to add
it in a non-default mod for people who want to mess with their game.

Side note, "x should be in because y is already in" is not an actual
argument, stop using it if you want me to take you seriously.

**Comment by chaosvolt** [comment]
Alright, no Sten crafting for us then. =w=

File: data/json/items/tools.json [hunk 3]
```diff
@@ -5818,6 +5869,30 @@
   },
 
   {
+    "id": "pliers",
+    "type": "TOOL",
+    "symbol": ";",
+    "color": "light_gray",
+    "name": "pliers",
+    "name_plural": "pliers",
+    "description": "This is a basic pair of slip-joint pliers, able to handle basic mechanical work. Anything to complex will require a wrench.",
```
Comment (Rivet-the-Zombie) [inline data/json/items/tools.json]: Double space after periods; typo: 'to' should be 'too'.

**Comment by chaosvolt** [comment]
Ack. @_@

**Comment by Zireael07** [comment]
Hmm, we have a craftable semi-auto gun atm?

+1 to making craftable auto guns a mod.

**Comment by chaosvolt** [comment]
As Kevin already said, the fact that a semi-auto gun is realistically more complex to craft does not set a valid precedent for crafting a full-auto gun.

And unrelated to this, that tube magazine comment in the data for it still makes my head hurt. :V

**Comment by Rivet-the-Zombie** [comment]
>that tube magazine comment in the data

Spire point ammunition in a tube magazine. I wonder what the person who added that one was thinking at the time.

**Comment by chaosvolt** [comment]
Exactly what I was wondering. o3o

**Comment by chaosvolt** [comment]
I will admit that it is a tad silly of me to snark at something that has no gameplay effect and isn't even visible unless you go json hunting, but still. ;w;

**Comment by DavidKeaton** [comment]
I have complained about silly things too. :-) (i.e. `auto usage` issue, anyone?)

**Comment by chaosvolt** [comment]
I blame being enough of a gun nut to notice inane things like that. o3o

**Comment by DavidKeaton** [comment]
I consider myself a gun nut, but not on you nor Rivet's scale.

**Comment by chaosvolt** [comment]
Rivet definitely seems to be the bigger gun nut though. Despite the extensive field manual collection I've gathered, the only guns I've had the chance to use are rather old hand-me-downs. ^^"

**Comment by chaosvolt** [comment]
Well, while I'm waiting on this, also PR'd a minor tileset update ( https://github.com/CleverRaven/Cataclysm-DDA/pull/12544 ).

Always leery of updating the master and starting a new branch when an older PR is still ongoing, but should be fine.

**Comment by Rivet-the-Zombie** [comment]
I can see some places were we can expand on this; stuff like a recipe where you collect a dozen normal screwdrivers and convert them into a screwdriver set and such, but I like where this is now.

**Comment by chaosvolt** [comment]
Ooh, thank you for the merge then.

And of course I'll hope the BAR turns out to be useful. Favorite as always. o3o

**Pull request merged**